src/remote/remote.ts

```diff
@@ -733,6 +765,95 @@ export class Remote {
 		return ` --log-dir ${escapeCommandArg(logDir)} -v`;
 	}
 
+	/**
+	 * Waits for an agent condition with progress notification and error handling.
+	 * Returns the updated agent or undefined if the user chooses to close remote.
+	 */
+	private async waitForAgentWithProgress(
+		monitor: WorkspaceMonitor,
+		agent: WorkspaceAgent,
+		options: {
+			progressTitle: string;
+			timeoutMs: number;
+			errorDialogTitle: string;
+		},
+		checker: (agent: WorkspaceAgent) => WorkspaceAgent | undefined,
```

Comment:
This threw me off a little, I think I expected a truthy value to mean the wait condition has been fulfilled and we should exit (like the callback to `filter` or `some` for example).

Also could this just return a boolean?  It looks like all we ever do is pass back the same agent that was passed in.